src/features/unified-share-modal/ContactsField.js

```diff
@@ -198,7 +200,12 @@ class ContactsField extends React.Component<Props, State> {
                 validator={validator}
             >
                 {contacts.map(({ email, text = null, id }) => (
-                    <ContactDatalistItem key={id} name={text} subtitle={email || groupLabel} />
+                    <ContactDatalistItem
+                        key={id}
+                        name={text}
+                        subtitle={email || groupLabel}
+                        title={email && email.length > MAX_NUM_CHAR_FOR_LISTITEM ? email : ''}
```

Comment:
You want to show browser tooltip or our own `<Tooltip>`?

Reply:
Yes, verified with design